pkg/handler/enabled_flag_validation.go

```diff
@@ -0,0 +1,201 @@
+	}
+
+	arr, ok := args[1].([]interface{})
+	if !ok {
+		return false, fmt.Errorf("arg 1 is not a slice! ***%v*** %v", args, reflect.TypeOf(args[1]))
+	}
+
+	functions := map[string]govaluate.ExpressionFunction{
+		"regexMatch": RegexMatchFunc,
+	}
+	expression, err := govaluate.NewEvaluableExpressionWithFunctions(function, functions)
+	if err != nil {
+		return false, err
+	}
+
+	found := false
+	for _, item := range arr {
+		params, ok := item.(map[string]interface{})
+		if !ok {
+			return false, fmt.Errorf("item is not a map %v", item)
+		}
+
+		result, err := expression.Evaluate(params)
+		if err != nil {
+			return false, err
+		}
+
+		if boolResult, ok := result.(bool); ok {
+			if boolResult {
+				found = true
+				break
+			}
+		}
+	}
+	return found, nil
+}
+
+func getEvaluationRules() ([]*govaluate.EvaluableExpression, error) {
+	createEvaluationRule.Do(func() {
+		for _, structExpressionString := range config.Config.EnabledFlagValidationRules {
+			functions := map[string]govaluate.ExpressionFunction{
+				"regexMatch": RegexMatchFunc,
+				"any":        Any,
+			}
+
+			expression, err := govaluate.NewEvaluableExpressionWithFunctions(structExpressionString, functions)
+			if expression == nil || err != nil {
+				evaluationRulesSetupError = err
+				return
+			}
+			evaluationRules = append(evaluationRules, expression)
+		}
+	})
+
+	return evaluationRules, evaluationRulesSetupError
+}
+
+func validateEnabledFlag(f *entity.Flag) (valid bool, err error) {
+	if config.Config.EnabledFlagValidationOperation == config.FlagValidationOperationAND {
+		valid = true
+	}
+
+	evaluationRules, err := getEvaluationRules()
+	if err != nil {
+		return false, err
+	}
+
+	for _, validationRule := range evaluationRules {
+		params := structs.Map(f)
+
+		result, err := validationRule.Evaluate(params)
+		if err != nil {
+			return false, err
+		}
+
+		resultBool, ok := result.(bool)
+		if !ok {
+			return false, fmt.Errorf("Unknown evaluation rule result type (%v)", result)
+		}
+
+		if config.Config.EnabledFlagValidationOperation == config.FlagValidationOperationOR {
+			valid = valid || resultBool
+			if valid {
+				break
+			}
+		} else if config.Config.EnabledFlagValidationOperation == config.FlagValidationOperationAND {
+			valid = valid && resultBool
+		} else {
+			return false, errors.New("Unknown FlagEnabledFlagValidationOperation - check server config")
+		}
+	}
+
+	return valid, nil
+}
+
+func validateFlagIfEnabled(f *entity.Flag) middleware.Responder {
+	if f.Enabled {
+		valid, err := validateEnabledFlag(f)
+		if err != nil {
+			return flag.NewPutFlagDefault(500).WithPayload(ErrorMessage("%s", err))
+		}
+		if !valid {
+			opMsg := "all"
+			if config.Config.EnabledFlagValidationOperation == config.FlagValidationOperationOR {
+				opMsg = "one"
+			}
+
+			return flag.NewPutFlagDefault(400).WithPayload(ErrorMessage("Flag failed %s of these validation rules: %s", opMsg, config.Config.EnabledFlagValidationRules))
```

Comment:
Is this necessarily true? Only a single condition has to fail in a series of "And" evaluations to fail the validation.